lib/src/rules/prefer_iterable_whereType.dart

```diff
@@ -80,8 +80,10 @@ class _Visitor extends SimpleAstVisitor<void> {
       expression = expression?.unParenthesized;
       if (expression is IsExpression && expression.notOperator == null) {
         final target = expression.expression;
+        var identifier = arg.parameters?.parameters.first.identifier;
         if (target is SimpleIdentifier &&
-            target.name == arg.parameters?.parameters.first.identifier!.name) {
+            identifier != null &&
+            target.name == identifier.name) {
```

Comment:
I think this would be cleaner as:

```dart
final target = expression.expression;
if (target is SimpleIdentifier &&
    target.name == arg.parameters?.parameters.first.identifier?.name) {
  rule.reportLint(node.methodName);
}
```

This way calculating the identifier can be short circuited and avoided if necessary, plus `name` isn't nullable if the right hand side evaluates to `null` the expression won't evaluate to true anyway.

Reply:
Nice.  Thanks!